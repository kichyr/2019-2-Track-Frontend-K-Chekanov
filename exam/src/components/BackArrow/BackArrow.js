import React from 'react';
// eslint-disable-next-line
import style from './backArrow.module.css';

function BackArrow() {
	return (
		<div className={'arrow'} style={{
			transform: 'scale(-1, 1) scale(0.3)'}}>
			<div className={'arrowTop'} />
			<div className={'arrowBottom'} />
		</div>
	);
}

export default BackArrow;