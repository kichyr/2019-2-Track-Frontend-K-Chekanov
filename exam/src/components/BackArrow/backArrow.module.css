
.arrow {
  cursor: pointer;
  position: relative;
  transform: translateX(-50%) translateY(-50%);
  transition: transform 0.1s;
  display: flex;
  width: 100%;
  height: 100%;
  align-items: center;
}
.arrowTop, .arrowBottom {
  background-color: rgb(248, 240, 240);
  height: 12px;
  left: -2px;
  position: absolute;
  width: 20vh;
  border-radius: 10px;
}
.arrowTop:after, .arrowBottom:after {
  background-color: rgb(129, 127, 207);
  content: "";
  height: 100%;
  position: absolute;
  top: 0;
  transition: all 0.15s;
}
.arrowTop {
  transform: rotate(45deg);
  transform-origin: bottom right;
}
.arrowTop:after {
  right: 0;
  transition-delay: 0s;
}
.arrowBottom {
  transform: rotate(-45deg);
  transform-origin: top right;
  border-radius: 10px;

}
.arrowBottom:after {
  left: 0;
  transition-delay: 0.15s;
}
.arrow:hover .arrowTop:after {
  left: 0;
  transition-delay: 0.15s;
}
.arrow:hover .arrowBottom:after {
  right: 0;
  transition-delay: 0s;
}
.arrow:active {
  transform: translateX(-50%) translateY(-50%) scale(0.9);
}
