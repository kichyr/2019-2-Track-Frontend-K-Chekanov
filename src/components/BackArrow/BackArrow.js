import React from 'react';
import style from './backArrow.module.css';

function BackArrow() {
	return (
		<div className={style.arrow} style={{
			transform: 'scale(-1, 1) scale(0.3)'}}>
			<div className={style.arrowTop} />
			<div className={style.arrowBottom} />
		</div>
	);
}

export default BackArrow;