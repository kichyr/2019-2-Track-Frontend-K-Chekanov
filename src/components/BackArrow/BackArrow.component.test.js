import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

import BackArrow from './BackArrow'

it('renders back arrow', () => {
  const output = shallow(<BackArrow />)
  expect(output).toMatchSnapshot()
})
