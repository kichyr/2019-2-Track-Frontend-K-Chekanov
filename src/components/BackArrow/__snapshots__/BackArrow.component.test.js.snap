// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders back arrow 1`] = `ShallowWrapper {}`;
